chain/ethereum/src/chain.rs

```diff
@@ -311,6 +313,20 @@ impl Blockchain for Chain {
             .await
     }
 
+    async fn final_block_pointer_for(
+        &self,
+        logger: &Logger,
+        block: &BlockFinality,
+    ) -> Result<BlockPtr, IngestorError> {
+        let final_block_number = match block.number() {
+            x if x >= self.reorg_threshold => x - self.reorg_threshold,
```

Comment:
I haven't wrapped around the whole PR yet. But generally one of the advantages of the firehose is to not have to configure a reorg threshold at all. Maybe this could just revert one block a time until it gets a consistent response from the firehose?

Reply:
So all this PR is around the fact that we want to start a subgraph from a particular block_ptr. So the subgraph is at block `#100 abc`. 

Now, we ask Firehose to start at block `#101` but we cannot specify the block hash we want to start from. So the rough idea is the following:

- Start Firehose with block `#101`
- Retrieve the first block sent by Firehose
- Get the parent from it
- Ensure the `parent` is == `#100 abc`
- If it's not the case, rewind the subgraph to the last final block of the block `#100`

This is of course an edge case that will not happen often. 

I like your idea to be honest, it's probably easier to implement than doing the full rewind multiple blocks behind.

Reply:
Well, now that I think about it, ideally I would revert to the parent block of the current subgraph block ptr, but I don't think I have such info, is it available? 

Let's discuss this at the meeting in ~1h.

Reply:
I've moved all the logic in the `FirehoseBlockStream` now. I still require getting the last final block where to revert to. But it's good.

I still need to update `revert_block_...` to actually accept reverting more than one block, and I need to also test the revertion code.